### ISSUE TITLE
Distinguish between workspaces

### ISSUE THREAD

**Issue by tildebox** [django-tenants/django-tenants#850]
This is not an issue. I'm trying to find a solution and hope to get an answer here.

Users can register on my website with an e-mail address. After logging in, users can create workspaces.

Team members can be invited to each workspace. The team members can only see the team members in this workspace.

The user can then switch between the workspaces.

For the separation, I've seen sites create a different subdomain for each workspace:

david.example.com
james-smith.example.com
lucy3.example.com

Are there other options? I saw on a website that they don't use anything like that in the URL or in the links. I wonder how they can tell which workspace the user is in. Can you give me some keywords so I can continue searching.

I would like to know if this is the right place for my project.

I also saw that there is a package django-tenant-schemas. Would that be more suitable for my project?

**Comment by rajeshr188** [comment]
yes associate a workspace with the user,create a middleware to switch schema based on request.user.workspace,provide a list of workspace(schemas) a user is member of upon selection set workspace of user to selected workspace middleware automatically update the schema,easy.

**Comment by tomturner** [comment]
Closing as part of a backlog tidy-up. This is a usage/integration question rather than a bug in django-tenants, and it has been inactive for some years — the versions and Django releases involved have moved on a long way since.

If you're still stuck, **[GitHub Discussions](https://github.com/django-tenants/django-tenants/discussions)** is the better home for questions like this and gets more eyes than the issue tracker.

And if you believe this *is* a bug in django-tenants, please reopen or open a fresh issue with the django-tenants and Django versions, a minimal `settings.py`, and the steps to reproduce — that's what makes it actionable.

Thanks for taking the time to write it up.
